### PR TITLE
[FS-13] - Added stripe integration on Subscription model

### PR DESCRIPTION
**Pull request by gleandroj**
## Description
Added stripe integration on Subscription model

## Tickets

[FS-13](https://flatirons.atlassian.net/browse/FS-13)

### CODE REVIEW

File: app/models/flatirons/saas/subscription.rb [hunk 1]
```diff
@@ -10,11 +10,27 @@ module Flatirons::Saas
...
+    validates :stripe_price_id, presence: true
     validates :status, presence: true, inclusion: { in: statuses }
+    validate :stripe_customer_id, on: :create
+
+    before_create :create_stripe_subscription, prepend: true
+
+    private
+
+    def stripe_customer_id
+      errors.add(:subscriptable, 'stripe_customer_id is not valid') unless subscriptable&.stripe_customer_id
```
Comment (animista01) [inline app/models/flatirons/saas/subscription.rb]: is not valid -> is required

File: app/models/flatirons/saas/subscription.rb [hunk 2]
```diff
@@ -10,11 +10,27 @@ module Flatirons::Saas
...
+    end
+
+    def create_stripe_subscription
+      subscription = stripe_service.create_subscription subscriptable.stripe_customer_id, stripe_price_id
```
Comment (animista01) [inline app/models/flatirons/saas/subscription.rb]: close methods params in parenthesis

**Review by animista01 — CHANGES_REQUESTED** [review]

File: app/models/flatirons/saas/subscription.rb (hunk 1 above)
Comment (gleandroj) [reply app/models/flatirons/saas/subscription.rb]: Done!

File: app/models/flatirons/saas/subscription.rb (hunk 2 above)
Comment (gleandroj) [reply app/models/flatirons/saas/subscription.rb]: Done!

**Review by animista01 — APPROVED** [review]

**Pull request merged**